WebApp/src/components/RegisterPage/RegisterPage.js

```diff
@@ -1,65 +1,220 @@
+      },
+      { 
+        field: 'surname', 
+        method: 'isEmpty', 
+        validWhen: false, 
+        message: 'Priezvisko je povinné.'
+      },
+      { 
+        field: 'password', 
+        method: 'isEmpty', 
+        validWhen: false, 
+        message: 'Heslo je povinné.'
+      },
+      { 
+        field: 'password', 
+        method: this.passwordLength,
+        validWhen: true, 
+        message: 'Heslo musí obsahovať aspoň 8 znakov.'
+      },
+      { 
+        field: 'password', 
+        method: 'matches',
+        args: [/\d/],
+        validWhen: true, 
+        message: 'Heslo musí obsahovať aspoň jednu číslicu.'
+      },
+      { 
+        field: 'password', 
+        method: 'matches',
+        args: [/.*[A-Z]/],
+        validWhen: true, 
+        message: 'Heslo musí obsahovať aspoň jedno veľké písmeno.'
+      },
+      { 
+        field: 'password', 
+        method: 'matches',
+        args: [/.*[a-z]/],
+        validWhen: true, 
+        message: 'Heslo musí obsahovať aspoň jedno malé písmeno.'
+      },
+      { 
+        field: 'passwordAgain', 
+        method: 'isEmpty', 
+        validWhen: false, 
+        message: 'Potvrdenie hesla je povinné.'
+       },
+      { 
+        field: 'passwordAgain', 
+        method: this.passwordMatch,
+        validWhen: true, 
+        message: 'Heslá sa nezhodujú.'
+      }
+    ]);
+
+    this.state = {
+      name: '',
+      surname: '',
+      email: '',
+      password: '',
+      passwordAgain: '',
+      validation: this.validator.valid(),
+      serverErrors: []
+    }
+  }
+
+  submitted = false;  
+  passwordMatch = (confirmation, state) => (state.password === confirmation)
+  passwordLength = (confirmation, state) => (state.password.length > 7)
+
+  handleInputChange = event => {
+    event.preventDefault();
+
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  }
 
   onSubmit = () => {
     const data = {
+      name: this.state.name,
+      surname: this.state.surname,
       email: this.state.email,
-      password: this.state.password
+      password: this.state.password,
+      passwordAgain: this.state.passwordAgain
     };
-    console.log(data);
-    axios({
-      method: 'post',
-      url: '/api/user/register',
-      data
-    })
-      .then(() => {
-        console.log('Api call succeeded');
-      })
-      .catch(() => {
-        console.error('Api call failed');
+    
+    const validation = this.validator.validate(this.state);
+    this.setState({ validation });
+    this.submitted = true;
+
+    if (validation.isValid) {
+      axios({
+        method: 'post',
+        url: '/api/user/register',
+        data,
+      }).then(() => {
+        this.props.history.push('/');
+      }).catch(error => {
+        this.setState({ serverErrors: new Array() })
+        var newErrorsArray = new Array();
+        var dictionary = error.response.data.errors;
+        for (var key in dictionary) {
```

Comment:
This for loop is so imperative. Checkout lodash, it could be written in one line: `const serverErrors = _.filter(_.values(dictionary))`

Reply:
fixed